pkg/registries/adapters/helm_adapter.go

```diff
@@ -0,0 +1,322 @@
+		var chartVersions []string
+		charts, ok := r.Charts[name]
+		if !ok {
+			continue
+		}
+
+		for _, chart := range charts {
+			chartVersions = append(chartVersions, chart.Version)
+		}
+		// Use the latest chart for creating the bundle
+		chart := charts[0]
+
+		resp, err := http.Get(chart.URLs[0])
+		if err != nil {
+			return specs, err
+		}
+		defer resp.Body.Close()
+
+		values := r.loadArchive(resp.Body)
+
+		// Convert chart to Bundle Spec
+		spec := &apb.Spec{
+			Runtime:     2,
+			Version:     "1.0",
+			Async:       "optional",
+			Bindable:    false,
+			Image:       r.Config.Runner,
+			FQName:      chart.Name,
+			Tags:        chart.Keywords,
+			Description: chart.Description,
+			Metadata: map[string]interface{}{
+				//"longDescription":  chart.Description,
+				"displayName":      fmt.Sprintf("%s (Helm)", chart.Name),
+				"documentationUrl": chart.Home,
+				"dependencies":     chart.Sources,
+				"imageUrl":         chart.Icon,
+			},
+			Plans: []apb.Plan{
+				apb.Plan{
+					Name:        "default",
+					Description: "Default plan for running helm charts",
+					Parameters: []apb.ParameterDescriptor{
+						apb.ParameterDescriptor{
+							Name:      "repo",
+							Title:     "Helm Chart Repository URL",
+							Type:      "string",
+							Default:   r.Config.URL.String(),
+							Pattern:   fmt.Sprintf("^%s$", r.Config.URL.String()),
+							Updatable: false,
+							Required:  false,
+						},
+						apb.ParameterDescriptor{
+							Name:      "chart",
+							Title:     "Helm Chart",
+							Type:      "string",
+							Default:   chart.Name,
+							Pattern:   fmt.Sprintf("^%s$", chart.Name),
+							Updatable: false,
+							Required:  false,
+						},
+						apb.ParameterDescriptor{
+							Name:      "version",
+							Title:     "Helm Chart Version",
+							Type:      "enum",
+							Enum:      chartVersions,
+							Default:   chart.Version,
+							Updatable: true,
+							Required:  false,
+						},
+						apb.ParameterDescriptor{
+							Name:        "values",
+							Title:       "Values",
+							Type:        "string",
+							DisplayType: "textarea",
+							Default:     values,
+							Updatable:   true,
+							Required:    false,
+						},
+					},
+				},
+			},
+		}
+
+		specs = append(specs, spec)
+	}
+
+	return specs, nil
+}
+
+// getHelmIndex returns an helm repository IndexFile object
+// https://github.com/kubernetes/helm/blob/48e703997016f3edeb4f0b90e6cfdb3456ce3db0/pkg/repo/index.go#L271
+func (r *HelmAdapter) getHelmIndex() (*IndexFile, error) {
+	index := &IndexFile{}
+
+	url := strings.TrimSuffix(r.Config.URL.String(), "/") + helmIndexPath
+	resp, err := http.Get(url)
+	if err != nil {
+		return index, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return index, err
+	}
+
+	err = yaml.Unmarshal(body, index)
+	if err != nil {
+		return index, err
+	}
+	for _, versions := range index.Entries {
+		sort.Sort(sort.Reverse(versions))
```

Comment:
I don't think that we need to sort twice

Reply:
I'm going to defer to your knowledge of Golang here, but this does appear in line with https://golang.org/pkg/sort/#Reverse.

Reply:
Looks like `sort.Sort` executes the sort built up on a `sort.Interface` wrapped type:

https://play.golang.com/

Note: It mutates the original slice, you aren't getting a copy back.